src/Microsoft.OData.Client/Serialization/PrimitiveXmlConverter.cs

```diff
@@ -858,10 +858,36 @@ internal override string ToString(object instance)
         }
     }
 
+    /// <summary>
+    /// Convert between primitive types Edm.Date and string
+    /// </summary>
+    internal sealed class DateOnlyTypeConverter : PrimitiveTypeConverter
+    {
+        /// <summary>
+        /// Create an instance of primitive type from a string representation
+        /// </summary>
+        /// <param name="text">The string representation</param>
+        /// <returns>An instance of primitive type</returns>
+        internal override object Parse(string text)
+        {
+            return PlatformHelper.ConvertStringToDateOnly(text);
+        }
+
+        /// <summary>
+        /// Convert an instance of primitive type to string
+        /// </summary>
+        /// <param name="instance">The instance</param>
+        /// <returns>The string representation of the instance</returns>
+        internal override string ToString(object instance)
+        {
+            return ((Date)(DateOnly)instance).ToString();

```

Comment:
Curious why the instance must be double-coerced before serializing, same with TimeOnly

Reply:
I mentioned in the only of changed file. If the 'instance' is 'DateOnly', but it's boxed since it's 'System.Object'.
When we 'cast' to 'System.Object', it fails since we cannot 'unbox' 'DateOnly' to 'Date' directly. 
I have to 'unbox' it to 'DateOnly', and cast to 'Date' implicitly.. #Resolved.